### PR TITLE
OP-1134 make gui auto find supporting font

### PR DESCRIPTION
**Pull request by mwithi**
See OP-1134.

### CODE REVIEW

File: src/main/java/org/isf/menu/gui/Menu.java [hunk 1]
```diff
@@ -104,6 +110,52 @@ public static boolean isAtLeastVersion(String requiredVersion) {
...
+
+		GeneralData.initialize();
+		MessageBundle.getBundle();
+		String textToCheck = MessageBundle.getMessage("angal.common.address.txt"); // Any string from bundle to check
+		LOGGER.debug("textToCheck: {}", textToCheck);
+
+		Font currentFont = UIManager.getFont("Label.font");
+
+		if (currentFont != null) {
+			LOGGER.debug("Current Font: " + currentFont.getFontName() + " " + currentFont.getSize());
```
Comment (dbmalkovsky) [inline src/main/java/org/isf/menu/gui/Menu.java:133]:
```suggestion
			LOGGER.debug("Current Font: { } { }", currentFont.getFontName(), currentFont.getSize());
```

File: src/main/java/org/isf/menu/gui/Menu.java [hunk 2]
```diff
@@ -104,6 +110,52 @@ public static boolean isAtLeastVersion(String requiredVersion) {
 		}
 	}
 
+	public static void setUIFont(javax.swing.plaf.FontUIResource f) {
```
Comment (dbmalkovsky) [inline src/main/java/org/isf/menu/gui/Menu.java:113]: the class should be imported and not fully qualivied

File: src/main/java/org/isf/menu/gui/Menu.java [hunk 3]
```diff
@@ -104,6 +110,52 @@ public static boolean isAtLeastVersion(String requiredVersion) {
...
+		Enumeration<Object> keys = UIManager.getDefaults().keys();
+		while (keys.hasMoreElements()) {
+			Object key = keys.nextElement();
+			Object value = UIManager.get(key);
+			if (value instanceof javax.swing.plaf.FontUIResource)
```
Comment (dbmalkovsky) [inline src/main/java/org/isf/menu/gui/Menu.java:118]: the class should be imported and not fully qualified

File: src/main/java/org/isf/menu/gui/Menu.java [hunk 4]
```diff
@@ -104,6 +110,52 @@ public static boolean isAtLeastVersion(String requiredVersion) {
...
+			LOGGER.debug("The current font supports the selected language.");
+		} else {
+			LOGGER.debug("The current font does not support the selected language.");
+
+			// Find a font that supports the provided textToCheck
+			Font[] availableFonts = GraphicsEnvironment.getLocalGraphicsEnvironment().getAllFonts();
+			for (Font font : availableFonts) {
+				if (font.canDisplayUpTo(textToCheck) == -1) {
+					LOGGER.debug("Found a font that supports the selected language: {}", font.getFontName());
+					setUIFont(new javax.swing.plaf.FontUIResource(font.getFontName(), Font.PLAIN, 12));
```
Comment (dbmalkovsky) [inline src/main/java/org/isf/menu/gui/Menu.java:150]: the class should be imported and not fully qualified in the code

Comment (mwithi) [reply src/main/java/org/isf/menu/gui/Menu.java]: right, thanks!

File: src/main/java/org/isf/menu/gui/Menu.java [hunk 5]
```diff
@@ -104,6 +111,52 @@ public static boolean isAtLeastVersion(String requiredVersion) {
...
+
+		GeneralData.initialize();
+		MessageBundle.getBundle();
+		String textToCheck = MessageBundle.getMessage("angal.common.address.txt"); // Any string from bundle to check
+		LOGGER.debug("textToCheck: {}", textToCheck);
+
+		Font currentFont = UIManager.getFont("Label.font");
+
+		if (currentFont != null) {
+			LOGGER.debug("Current Font: { } { }", currentFont.getFontName(), currentFont.getSize());
```
Comment (dbmalkovsky) [inline src/main/java/org/isf/menu/gui/Menu.java:134]:
```suggestion
			LOGGER.debug("Current Font: {} {}", currentFont.getFontName(), currentFont.getSize());
```
No spaces

**Review by dbmalkovsky — APPROVED** [review]
LGTM; thanks

**Pull request merged**